sql/core/src/test/scala/org/apache/spark/sql/execution/benchmark/DataSourceReadBenchmark.scala

```diff
@@ -119,31 +119,36 @@ object DataSourceReadBenchmark extends SqlBasedBenchmark {
 
         prepareTable(dir, spark.sql(s"SELECT CAST(value as ${dataType.sql}) id FROM t1"))
 
+        val query = dataType match {
```

Comment:
Why is this change required?

Reply:
Spark SQL does not support `sum(id)` when `id` is `Boolean`. It throws an Exception. Hence, `sum(cast(id as bigint))` is required for the `BooleanType` benchmark.